project/apps/user-service/src/adapters/db/users.supabase.ts

```diff
@@ -21,5 +28,166 @@ export class SupabaseUsersRepository implements UsersRepository {
     this.supabase = createClient(supabaseUrl, supabaseKey);
   }
 
-  // Refer to questions.supabase.ts for examples
+  async findAll(filters: UserFiltersDto): Promise<UserCollectionDto> {
+    const { email, username, offset, limit } = filters;
+
+    let queryBuilder = this.supabase
+      .from(this.PROFILES_TABLE)
+      .select('*', { count: 'exact' });
+
+    if (email || username) {
+      const orFilter = [];
+      if (email) {
+        orFilter.push(`email.eq.${email}`);
+      }
+      if (username) {
+        orFilter.push(`username.eq.${username}`);
+      }
+      if (orFilter.length > 0) {
+        queryBuilder = queryBuilder.or(orFilter.join(','));
+      }
+    }
+
+    const totalCountQuery = queryBuilder;
+
+    let dataQuery = queryBuilder;
+    if (limit) {
+      if (offset) {
+        dataQuery = dataQuery.range(offset, offset + limit - 1); // Supabase range is inclusive
+      } else {
+        dataQuery = dataQuery.limit(limit);
+      }
+    }
+
+    // Execute the data query
+    const { data: users, error } = await dataQuery;
+
+    // Execute the total count query
+    const { count: totalCount, error: totalCountError } = await totalCountQuery;
+
+    if (error || totalCountError) {
+      throw error || totalCountError;
+    }
+
+    const metadata: collectionMetadataDto = {
+      count: users ? users.length : 0,
+      totalCount: totalCount ?? 0,
+    };
+
+    return {
+      metadata,
+      users,
+    } as UserCollectionDto;
+  }
+
+  async findById(id: string): Promise<UserDataDto> {
+    const { data, error } = await this.supabase
+      .from(this.PROFILES_TABLE)
+      .select()
+      .eq('id', id)
+      .single<UserDataDto>();
+
+    if (error) {
+      throw error;
+    }
+
+    return data;
+  }
+
+  async updateById(userDetails: UpdateUserDto): Promise<UserDataDto> {
+    const {
+      id,
+      email: newEmail,
+      username: newUsername,
+      password: newPassword,
+    } = userDetails;
+
+    // Update user email and/or password in auth table
+    const { error: authError } = await this.supabase.auth.updateUser({
+      email: newEmail,
+      password: newPassword,
+    });
```

Comment:
```suggestion
    // Update user email and/or password in auth table
    const { error: authError } = await this.supabase.auth.updateUser({
      email: newEmail,
      password: newPassword,
      data: { username: newUsername },
    });
```

also update the username field in data, like in:
https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g30/blob/220e8a6e5c21d3273a221b7ba7d71db2a2329aba/project/apps/auth-service/src/adapters/db/auth.supabase.ts#L78-L81